marklogic-data-hub-central/ui/src/components/flows/flows.tsx

```diff
@@ -446,47 +448,84 @@ const Flows: React.FC<Props> = (props) => {
     }
 
-    let data = {stepName: "", stepNumber: -1, stepDefinitionType: "", isChecked: false, flowName: "", stepId: "", sourceFormat: ""};
-    data.stepName = checkedValues;
-    data.stepNumber = stepNumber;
-    data.stepDefinitionType = stepDefinitionType;
-    data.isChecked = event !== "default" ? event.target.checked : true;
-    data.flowName = flowNames;
-    data.stepId = stepId;
-    data.sourceFormat = sourceFormat;
+    if (fromCheckAll) {
+      if (checkAll) {
+
+        let stepsSelectedFlow = selectedStepDetails.filter(function (obj) {
+          return obj.flowName === flowNames;
+        });
 
-    let obj = selectedStepDetails;
+        stepsSelectedFlow.map((obj) => {
+          let selectedStepOptionsAux = selectedStepOptions;
+          selectedStepOptionsAux[flowNames + "_" + obj.stepName + "_" + obj.stepNumber] = false;
+          setSelectedStepOptions(selectedStepOptionsAux);
+        });
 
-    if (data.isChecked) {
-      obj.push(data);
+        if (currentFlowName.length > 0) {
+          let selectedStepDetailsAux = selectedStepDetails.filter(function (obj) {
+            return obj.flowName !== flowNames && obj.stepName !== checkedValues;
+          });
+          setSelectedStepDetails(selectedStepDetailsAux);
+          selectedStepDetails.shift();
+        }
+
+        let arrayLoadChecksStepsAux = arrayLoadChecksSteps;
+        arrayLoadChecksStepsAux.map(function (obj) {
+          if (obj.flowName === flowNames) obj.checked = false;
+        });
+        setArrayLoadChecksSteps(arrayLoadChecksStepsAux);
+      } else {
+        {
+          props.flows.map((flow) => (
+            flow["name"] === flowNames &&
+            flow?.steps && flow.steps.map((step) => {
+              controlsCheckboxes(step, step.stepDefinitionType, flow.name);
+            })));
+        }
+      }
     } else {
-      for (let i = 0; i < obj.length; i++) {
-        if (obj[i].stepName === checkedValues) {
-          obj.splice(i, 1);
+      let originRender = event !== "default" ? true : false;
+      let data = {stepName: "", stepNumber: -1, stepDefinitionType: "", isChecked: false, flowName: "", stepId: "", sourceFormat: ""};
+      data.stepName = checkedValues;
+      data.stepNumber = stepNumber;
+      data.stepDefinitionType = stepDefinitionType;
+      data.isChecked = originRender ? event.target.checked : true;
+      data.flowName = flowNames;
+      data.stepId = stepId;
+      data.sourceFormat = sourceFormat;
+
+      let obj = selectedStepDetails;
+      if (data.isChecked) {
+        obj.push(data);
+      } else {
+        for (let i = 0; i < obj.length; i++) {
+          if (obj[i].stepName === checkedValues) {
+            obj.splice(i, 1);
+          }
         }
       }
-    }
 
-    if (event !== "default" && stepDefinitionType === "ingestion") {
-      handleArrayLoadChecksSteps(flowNames, checkedValues, stepNumber);
-    }
+      if (originRender && stepDefinitionType === "ingestion") {
```

Comment:
You could add the .toLowerCase() to stepDefinitionType to avoid the error we talked about yesterday about stepTypes showing in all caps and not passing the comparisson

Reply:
Good catch @lucianaabc-ml  I added this in all the places related with "ingestion".